src/receive_imf.rs

```diff
@@ -1014,15 +1014,16 @@ async fn add_parts(
             }
         }
 
-        state = if (seen && replace_msg_id.is_none())
+        state = if replace_msg_id.is_some() {
```

Comment:
Not sure that the code can be rewritten this way. If a message is replaced, that doesn't mean it's already noticed. Maybe it's replaced because `DownloadLimit` was changed and the message was resent. But i didn't test such a scenario, so i decided not to change the code here too much